src/neo4j_graphrag/experimental/pipeline/kg_builder.py

```diff
@@ -0,0 +1,214 @@
+            pdf_loader=self.pdf_loader,
+            on_error=on_error,
+        )
+
+        self.llm = self.config.llm
+        self.driver = self.config.driver
+        self.file_path = self.config.file_path
+        self.text = self.config.text
+        self.text_splitter = self.config.text_splitter or FixedSizeSplitter()
+        self.on_error = self.config.on_error
+
+        self.pipeline = self._build_pipeline()
+
+    def _build_pipeline(self) -> Pipeline:
+        pipe = Pipeline()
+
+        if self.file_path:
+            pipe.add_component(self.pdf_loader, "loader")
+            loader_inputs = {"filepath": self.file_path}
+            document_info_input = "loader.document_info"
+        else:
+            loader_inputs = {}
+            document_info_input = {"path": "direct_text_input"}
+
+        pipe.add_component(
+            self.text_splitter,
+            "splitter",
+        )
+
+        pipe.add_component(SchemaBuilder(), "schema")
+        pipe.add_component(
+            LLMEntityRelationExtractor(llm=self.llm, on_error=self.on_error),
+            "extractor",
+        )
+        pipe.add_component(self.kg_writer, "writer")
+
+        if self.file_path:
+            pipe.connect(
+                "loader",
+                "splitter",
+                input_config={"text": "loader.text"},
+            )
+            self.pipe_inputs = {}
+        else:
+            self.pipe_inputs = {"splitter": {"text": self.text}}
+
+        pipe.connect(
+            "splitter",
+            "extractor",
+            input_config={"chunks": "splitter"},
+        )
+        pipe.connect(
+            "schema",
+            "extractor",
+            input_config={
+                "schema": "schema",
+                "document_info": document_info_input,
+            },
+        )
+        pipe.connect(
+            "extractor",
+            "writer",
+            input_config={"graph": "extractor"},
+        )
+
+        self.pipe_inputs.update(
+            {
+                "schema": {
+                    "entities": self.entities,
+                    "relations": self.relations,
+                    "potential_schema": self.potential_schema,
+                },
+            }
+        )
+
+        if self.file_path:
+            self.pipe_inputs["loader"] = loader_inputs
+        else:
+            self.pipe_inputs["extractor"] = {"document_info": document_info_input}
+
+        return pipe
+
+    async def run_async(self) -> PipelineResult:
+        """
+        Asynchronously runs the knowledge graph building process.
+
+        Returns:
+            PipelineResult: The result of the pipeline execution.
+        """
+        return await self.pipeline.run(self.pipe_inputs)
+
+    def run(self) -> PipelineResult:
+        """
+        Runs the knowledge graph building process.
+
+        Returns:
+            PipelineResult: The result of the pipeline execution.
+        """
+        return asyncio.run(self.run_async())
```

Comment:
Note that this won't work when an event loop is already running, e.g. in Jupyter (but you can just use `await .run()
` in these caess.